src/explore-education-statistics-ckeditor/plugins/comments/commentsEditing.js

```diff
@@ -0,0 +1,351 @@
+            )
+          ) {
+            this.config.undoRedoComment(
+              'undoResolveComment',
+              lastOperation.name,
+            );
+            return;
+          }
+
+          this.removedMarkers = this.removedMarkers.filter(
+            marker => marker !== lastOperation.name,
+          );
+          this.config.undoRedoComment('undoRemoveComment', lastOperation.name);
+          return;
+        }
+
+        if (
+          lastOperation.name &&
+          lastOperation.name === 'commentplaceholder' &&
+          lastOperation.newRange &&
+          !lastOperation.oldRange
+        ) {
+          this.config.undoRedoComment(
+            'undoAddComment',
+            lastOperation.batch.operations[0].name,
+          );
+          return;
+        }
+
+        if (
+          lastOperation.name &&
+          lastOperation.name.startsWith('resolvedcomment:')
+        ) {
+          this.config.undoRedoComment(
+            'undoUnresolveComment',
+            lastOperation.name,
+          );
+        }
+      });
+
+      /**
+       * Watch for redo operations we need to handle.
+       */
+      editor.commands.get('redo').on('execute', () => {
+        const docVersion = editor.model.document.version;
+        const lastOperation = editor.model.document.history.getOperation(
+          docVersion - 1,
+        );
+
+        if (lastOperation.name && lastOperation.name.startsWith('comment:')) {
+          if (
+            lastOperation.batch.operations[0].name &&
+            lastOperation.batch.operations[0].name.startsWith(
+              'resolvedcomment:',
+            )
+          ) {
+            this.config.undoRedoComment(
+              'redoUnresolveComment',
+              lastOperation.name,
+            );
+            return;
+          }
+
+          if (
+            lastOperation.batch.operations[0].name &&
+            lastOperation.batch.operations[0].name === 'commentplaceholder'
+          ) {
+            this.config.undoRedoComment('redoAddComment', lastOperation.name);
+            return;
+          }
+
+          this.config.undoRedoComment('redoRemoveComment', lastOperation.name);
+          return;
+        }
+
+        if (
+          lastOperation.name &&
+          lastOperation.name.startsWith('resolvedcomment:')
+        ) {
+          this.config.undoRedoComment('redoResolveComment', lastOperation.name);
+        }
+      });
+    });
+
+    /**
+     * Remove the marker when the remove comment command is executed.
+     */
+    this.listenTo(removeCommentCommand, 'execute', () => {
+      if (editor.model.markers.has(removeCommentCommand.commentName)) {
+        editor.model.change(writer =>
+          writer.removeMarker(removeCommentCommand.commentName),
+        );
+      }
+    });
+
+    /**
+     * Set the selected comment and update the marker when the select comment command is executed.
+     */
+    this.listenTo(selectCommentCommand, 'execute', () => {
```

Comment:
I tried to move this to the command but I couldn't get it to work because selectedComment needs to be set and available in the editingDowncast conversions in CommentsEditing before the marker is updated.

Reply:
Fair enough! Will defer to you on the matter as I don't really understand most of what's going on in the CKEditor code anyway :smile: